capsules/extra/src/bmi270.rs

```diff
@@ -0,0 +1,520 @@
+    i2c: &'a I,
+    alarm: &'a A,
+    ninedof_client: OptionalCell<&'a dyn NineDofClient>,
+    state: Cell<State>,
+    pending_gyro: Cell<bool>,
+    pending_accel: Cell<bool>,
+}
+
+impl<'a, A: Alarm<'a>, I: I2CDevice> BMI270<'a, A, I> {
+    pub fn new(
+        i2c_bus: &'a I,
+        alarm: &'a A,
+        buffer: &'static mut [u8],
+        config_file: &'static mut [u8],
+    ) -> BMI270<'a, A, I> {
+        BMI270 {
+            buffer: TakeCell::new(buffer),
+            config_file: TakeCell::new(config_file),
+            i2c: i2c_bus,
+            alarm: alarm,
+            ninedof_client: OptionalCell::empty(),
+            state: Cell::new(State::Sleep),
+            pending_gyro: Cell::new(false),
+            pending_accel: Cell::new(false),
+        }
+    }
+
+    pub fn start_measurement(&self) -> Result<(), ErrorCode> {
+        self.buffer
+            .take()
+            .map(|buffer| {
+                self.i2c.enable();
+                match self.state.get() {
+                    State::Sleep => {
+                        buffer[0] = Registers::PwrConf as u8;
+                        buffer[1] = 0x00_u8;
+
+                        if let Err((_error, buffer)) = self.i2c.write(buffer, 2) {
+                            self.buffer.replace(buffer);
+                            self.i2c.disable();
+                        } else {
+                            self.state.set(State::WaitingForAlarm(450));
+                        }
+                    }
+                    State::Idle => {
+                        buffer[0] = Registers::Data8 as u8;
+
+                        if let Err((i2c_err, buffer)) = self.i2c.write(buffer, 1) {
+                            self.state.set(State::Sleep);
+                            self.buffer.replace(buffer);
+                            self.ninedof_client
+                                .map(|client| client.callback(i2c_err as usize, 0, 0));
+                        } else {
+                            self.state.set(State::Read);
+                        }
+                    }
+                    _ => {}
+                }
+            })
+            .ok_or(ErrorCode::FAIL)
+    }
+
+    fn handle_alarm(&self) {
+        let _ = self.buffer.take().map(|buffer| match self.state.get() {
+            State::WaitingForAlarm(us) => {
+                if us == 450 {
+                    buffer[0] = Registers::InitCtrl as u8;
+                    buffer[1] = 0x00_u8;
+
+                    if let Err((i2c_err, buffer)) = self.i2c.write(buffer, 2) {
+                        self.state.set(State::Sleep);
+                        self.buffer.replace(buffer);
+                        self.ninedof_client
+                            .map(|client| client.callback(i2c_err as usize, 0, 0));
+                    } else {
+                        self.state.set(State::InitWriteConfig);
+                    }
+                } else if us == 20000 {
```

Comment:
The times here and above should probably be a constant in the file (since they're used multiple places)